lib/SILGen/SILGenFunction.cpp

```diff
@@ -510,42 +510,49 @@ SILGenFunction::emitClosureValue(SILLocation loc, SILDeclRef constant,
 void SILGenFunction::emitFunction(FuncDecl *fd) {
   MagicFunctionName = SILGenModule::getMagicFunctionName(fd);
 
-  if (fd->isDistributedActorFactory()) {
-    emitDistributedActorFactory(fd);
-    return;
-  }
-
   auto captureInfo = SGM.M.Types.getLoweredLocalCaptures(SILDeclRef(fd));
   emitProfilerIncrement(fd->getTypecheckedBody());
   emitProlog(captureInfo, fd->getParameters(), fd->getImplicitSelfDecl(), fd,
              fd->getResultInterfaceType(), fd->hasThrows(), fd->getThrowsLoc());
   prepareEpilog(true, fd->hasThrows(), CleanupLocation(fd));
 
-  if (llvm::any_of(
-          *fd->getParameters(),
-          [](ParamDecl *p){ return p->hasAttachedPropertyWrapper(); })) {
-    // If any parameters have property wrappers, emit the local auxiliary
-    // variables before emitting the function body.
-    LexicalScope BraceScope(*this, CleanupLocation(fd));
-    for (auto *param : *fd->getParameters()) {
-      param->visitAuxiliaryDecls([&](VarDecl *auxiliaryVar) {
-        SILLocation WrapperLoc(auxiliaryVar);
-        WrapperLoc.markAsPrologue();
-        if (auto *patternBinding = auxiliaryVar->getParentPatternBinding())
-          visitPatternBindingDecl(patternBinding);
-
-        visit(auxiliaryVar);
-      });
-    }
-
-    emitStmt(fd->getTypecheckedBody());
+  if (fd->isDistributedActorFactory()) {
+    // Synthesize the factory function body
+    emitDistributedActorFactory(fd);
   } else {
-    emitStmt(fd->getTypecheckedBody());
+    // Emit the actual function body as usual
+    if (llvm::any_of(
+            *fd->getParameters(),
+            [](ParamDecl *p){ return p->hasAttachedPropertyWrapper(); })) {
+      // If any parameters have property wrappers, emit the local auxiliary
+      // variables before emitting the function body.
+      LexicalScope BraceScope(*this, CleanupLocation(fd));
+      for (auto *param : *fd->getParameters()) {
+        param->visitAuxiliaryDecls([&](VarDecl *auxiliaryVar) {
+          SILLocation WrapperLoc(auxiliaryVar);
+          WrapperLoc.markAsPrologue();
+          if (auto *patternBinding = auxiliaryVar->getParentPatternBinding())
+            visitPatternBindingDecl(patternBinding);
+
+          visit(auxiliaryVar);
+        });
+      }
+
+      emitStmt(fd->getTypecheckedBody());
+    } else {
+      emitStmt(fd->getTypecheckedBody());
+    }
   }
 
   emitEpilog(fd);
 
   mergeCleanupBlocks();
```

Comment:
and we must not just `return` since we want the Epilog :) 
Alternatively we could do those things inside the custom impl... but reusing seemed safer.